reactor-core/src/main/java/reactor/core/publisher/FluxPublishOn.java

```diff
@@ -893,12 +874,10 @@ void runSync() {
 			int missed = 1;
 
 			final ConditionalSubscriber<? super T> a = actual;
-			final Queue<T> q = queue;
+			final Queue<T> q = qs;
 
 			long e = produced;
-
 			for (; ; ) {
-
 				long r = requested;
```

Comment:
shouldn't this be `requested & Long.MAX_VALUE` ?

Reply:
Yes, it needs to cast requested here if the drain is called before the request